python/GafferUI/EventLoop.py

```diff
@@ -198,6 +198,25 @@ def removeIdleCallback( cls, callback ) :
 
 		cls.__idleCallbacks.remove( callback )
 
+	## Convenience method to introduce a delay on the mainEventLoop().
+	@classmethod
+	def waitForIdle( cls, count = 1000 ) :
```

Comment:
I'm not sure I buy that its a hack. Its necessary in a handful of places, two of which you named, the third of which is in the comment below. Or if I address that comment, in every single call site for `WidgetAlgo.grab()`. Or if I address this comment, there won't be any call sites for `WidgetAlgo.grab()` as of today, because they'd all use `screengrabapp.grab()` which wraps it up and includes the pause.

PS. `EventLoopTest.waitForIdle()` is passing on OSX now I've added 9ac50ae (its the dodgy `SceneViewTest.testFrame()` that's failed this time).